02-frequency_filter.py

```diff
@@ -63,34 +75,53 @@ def run_filter(subject, run=None, session=None):
     raw.load_data()
 
     # Band-pass the data channels (MEG and EEG)
-    msg = f'Filtering data between {config.l_freq} and {config.h_freq} (Hz)'
+    msg = (f'Filtering experimental data between {config.l_freq} and '
+           f'{config.h_freq} Hz')
     logger.info(gen_log_message(message=msg, step=2, subject=subject,
-                                session=session, run=run,))
+                                session=session, run=run))
 
-    raw.filter(config.l_freq, config.h_freq,
-               l_trans_bandwidth=config.l_trans_bandwidth,
-               h_trans_bandwidth=config.h_trans_bandwidth,
-               filter_length='auto', phase='zero', fir_window='hamming',
-               fir_design='firwin'
-               )
+    filter_kws = dict(l_freq=config.l_freq, h_freq=config.h_freq,
+                      l_trans_bandwidth=config.l_trans_bandwidth,
+                      h_trans_bandwidth=config.h_trans_bandwidth,
+                      filter_length='auto', phase='zero', fir_window='hamming',
+                      fir_design='firwin')
+    raw.filter(**filter_kws)
+
+    if config.noise_cov == 'emptyroom':
+        msg = f'Filtering empty-room recording.'
+        logger.info(gen_log_message(message=msg, step=2, subject=subject,
+                                    session=session, run=run,))
+        raw_er = mne.io.read_raw_fif(raw_er_fname_in)
+        raw_er.load_data()
+        raw_er.filter(**filter_kws)
 
     if config.resample_sfreq:
-        msg = f'Resampling data to {config.resample_sfreq:.1f} Hz'
+        msg = f'Resampling experimental data to {config.resample_sfreq:.1f} Hz'
         logger.info(gen_log_message(message=msg, step=2, subject=subject,
                                     session=session, run=run,))
         raw.resample(config.resample_sfreq, npad='auto')
 
+        if config.noise_cov == 'emptyroom':
+            msg = f'Resampling empty-room recording.'
+            logger.info(gen_log_message(message=msg, step=2, subject=subject,
+                                        session=session, run=run,))
+            raw_er.resample(config.resample_sfreq, npad='auto')
+
     raw.save(raw_fname_out, overwrite=True)
+    if config.noise_cov == 'emptyroom':
+        raw_er.save(raw_er_fname_out, overwrite=True)
 
     if config.interactive:
-        # plot raw data
+        # Plot raw data and power spectral density.
         raw.plot(n_channels=50, butterfly=True)
+        raw.plot_psd()
 
-        # plot power spectral densitiy
-        raw.plot_psd(area_mode='range', tmin=10.0, tmax=100.0,
-                     fmin=0., fmax=50., average=True)
+        if config.noise_cov == 'emptyroom':
+            raw_er.plot(n_channels=50, butterfly=True)
+            raw_er.plot_psd()
```

Comment:
I've never seen anyone plot empty room raw data to inspect it. Maybe @agramfort has different opinions.